### PR TITLE
Bump rke to 1.4.16 to support kubernetes 1.27

### PR DESCRIPTION
**Pull request by enrico9034**
## Issue: #443 

Need to support k8s 1.27 because 1.26 is EOL at the end of February 2024  (https://github.com/rancher/terraform-provider-rke/issues/443)

## Solution

Update rke package to v1.4.16

## Testing

`make test` passes.

### Automated Testing
* Test types added/modified:
    * Unit
    * None
* If "None" - Reason: only bump the rke provider
* If "None" - GH Issue: #443 

### Regressions Considerations

Existing / newly added automated tests that provide evidence there are no regressions.

### CODE REVIEW

**Comment by enrico9034** [comment]
Please @a-blender @jiaqiluo @felipe-colussi

**Comment by enrico9034** [comment]
Any update?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
+## 1.4.4 (March 6, 2024)
```
Comment (jiaqiluo) [inline CHANGELOG.md:1]: Can you remove this section for now?  It will be added when we are preparing the release.

File: go.mod [hunk 2]
```diff
@@ -7,7 +7,7 @@ require (
 	github.com/ghodss/yaml v1.0.0
 	github.com/hashicorp/go-version v1.6.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.29.0
-	github.com/rancher/rke v1.4.11
+	github.com/rancher/rke v1.4.15
```
Comment (jiaqiluo) [inline go.mod:10]: can you bump rke to 1.4.16 which is the latest?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
Thank you for raising the PR!  
I left a few comments.

File: go.mod (hunk 2 above)
Comment (jiaqiluo) [reply go.mod:10]: Can you also update the PR's title and description to use RKE 1.4.16?

File: CHANGELOG.md (hunk 1 above)
Comment (enrico9034) [reply CHANGELOG.md]: Done

File: go.mod (hunk 2 above)
Comment (enrico9034) [reply go.mod]: Done, I updated it in the PR title and in the code

**Review by jiaqiluo — APPROVED** [review]
LGTM. 
Thank you for the fix!

**Comment by jiaqiluo** [comment]
We need one more approval to merge the PR, then to cut a RC.

**Comment by raelix** [comment]
when the tag will be released?

**Pull request merged**